### PR TITLE
doc(sugar): added description and examples to dup

### PR DESCRIPTION
**Pull request by luismanfroni**
## Solving issue #14405

Added more description and examples to sugar/dup documentation.

#### Before:
image

#### After:
image

#### Before:
image

#### After
image

### CODE REVIEW

**Comment by ringabout** [comment]
Hi @luismanfroni
Could you close and reopen this pr to rerun CI?

**Comment by Araq** [comment]
Expanding the explanation for "dup" is nice but I think even more examples overdo it.

File: lib/pure/sugar.nim [hunk 1]
```diff
@@ -234,6 +242,21 @@ since (1, 1):
...
+      var c = "xyz"
+
+      # An underscore (_) can be used to denote the place of the argument you're passing:
+      # b = "xyz"
+      var d = dup c:
+        makePalindrome # xyzyx
+        sort(_, SortOrder.Descending) # zyyxx
+        makePalindrome # zyyxxxyyz
+
+      doAssert e == "zyyxxxyyz"
```
Comment (Varriount) [inline lib/pure/sugar.nim]: This appears to be using an undeclared variable

File: lib/pure/sugar.nim [hunk 2]
```diff
@@ -217,7 +217,15 @@ since (1, 1):
 
   macro dup*[T](arg: T, calls: varargs[untyped]): T =
     ## Turns an `in-place`:idx: algorithm into one that works on
-    ## a copy and returns this copy.
+    ## a copy and returns this copy, without modifying its input.
+    ##
+    ## The idea is that future versions of Nim will offer only
```
Comment (Varriount) [inline lib/pure/sugar.nim]: I'm not sure how appropriate it is to mention standard library motives in function documentation, especially when that motive hasn't been codified anywhere dedicated and explicit.

**Review by Varriount — CHANGES_REQUESTED** [review]

File: lib/pure/sugar.nim (hunk 1 above)
Comment (luismanfroni) [reply lib/pure/sugar.nim]: My bad, did push without testing locally.

File: lib/pure/sugar.nim (hunk 2 above)
Comment (luismanfroni) [reply lib/pure/sugar.nim]: Ok, removed the second part from the description. Thanks for the feedback!

**Comment by luismanfroni** [comment]
> Expanding the explanation for "dup" is nice but I think even more examples overdo it.

Changed doc to only have the underscore example.

**Pull request merged**